File: src/Backend/Models/EntityModels/DataObject.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models.EntityModels
{
    public class DataObject : Entity
    {
        public DataObject() { }

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int DataObjectId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        public ICollection<ObjectFormat> ObjectFormats { get; set; }
        public ICollection<ObjectTarget> ObjectTargets { get; set; }
    }
}

File: src/Frontend/src/app/data-object/data-object.ts
export interface DataObject {
    dataObjectId?: number;
    name: string;
    description: string;
    objectFormats: ObjectFormat[];
    objectTargets: ObjectTarget[];
}

export interface ObjectFormat {
    objectFormatId?: number;
}

export interface ObjectTarget {
    objectTargetId?: number;
    name: string;
    description: string;
    apiInfos: any[];
    fileInfos: any[];
    checkTypes: boolean;
    evalConditions: boolean;
    stopOnError: boolean;
}

export class DataObject implements DataObject {
    constructor(
        
    ) { }
}

export class ObjectFormat implements ObjectFormat {
    constructor(
        
    ) { }
}

export class ObjectTarget implements ObjectTarget {
    constructor(
        
    ) { }
}